Kudu.FunctionalTests/OneDeployTests.cs

```diff
@@ -0,0 +1,389 @@
+                response1.EnsureSuccessStatusCode();
+
+                var file2 = CreateRandomTestFile();
+                var response2 = await DeployNonZippedArtifact(appManager, file2, new ZipDeployMetadata(), "static", "iterative.txt");
+                response2.EnsureSuccessStatusCode();
+
+                await AssertSuccessfulDeploymentByFilenames(appManager, new string[] { "app.war", "iterative.txt" });
+            });
+        }
+
+
+
+        private static async Task<HttpResponseMessage> DeployNonZippedArtifact(
+            ApplicationManager appManager,
+            TestFile file,
+            ZipDeployMetadata metadata,
+            string type = null,
+            string path = null,
+            string async = "false")
+        {
+            TestTracer.Trace("Deploying file");
+            using (var fileStream = CreateFileStream(file))
+            {
+                IList<KeyValuePair<string, string>> queryParams = new List<KeyValuePair<string, string>>();
+                if (!string.IsNullOrWhiteSpace(type))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("type", type));
+                }
+                if (!string.IsNullOrWhiteSpace(path))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("path", path));
+                }
+                if (async != "false")
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("async", async));
+                }
+                return await appManager.OneDeployManager.PushDeployFromStream(fileStream, metadata, queryParams);
+            }
+        }
+
+        private static async Task<HttpResponseMessage> DeployZippedArtifact(ApplicationManager appManager,
+            TestFile[] files,
+            ZipDeployMetadata metadata,
+            string type = null,
+            string path = null,
+            string async = "false")
+        {
+            TestTracer.Trace("Deploying zip");
+            using (var zipStream = CreateZipStream(files))
+            {
+                IList<KeyValuePair<string, string>> queryParams = new List<KeyValuePair<string, string>>();
+                if (!string.IsNullOrWhiteSpace(type))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("type", type));
+                }
+                if (!string.IsNullOrWhiteSpace(path))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("path", path));
+                }
+                if (async != "false")
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("async", async));
+                }
+
+                return await appManager.OneDeployManager.PushDeployFromStream(
+                    zipStream,
+                    metadata,
+                    queryParams);
+            }
+        }
+
+        private static TestFile CreateRandomTestFile()
```

Comment:
All the code after this point seems to be the same as that in ZipDeploymentTests.cs. Move that to DeploymentTestHelper.cs, convert private to public and just call that from here (and from ZipDeploymentTests.cs)